src/cli/src/error.rs

```diff
@@ -40,6 +39,28 @@ pub enum CliError {
     Other(String),
 }
 
+#[derive(thiserror::Error, Debug)]
+pub enum ConsumerError {
+    #[error(transparent)]
+    IoError(#[from] IoError),
+    #[error(transparent)]
+    OutputError(#[from] OutputError),
+    #[error("Fluvio client error")]
+    ClientError(#[from] FluvioError),
+    #[error("Invalid argument: {0}")]
+    InvalidArg(String),
+    #[error("Error finding executable")]
+    WhichError(#[from] which::Error),
+    #[error("Unknown error: {0}")]
+    Other(String),
+}
+
+impl ConsumerError {
```

Comment:
Use From Trait

Reply:
From what? CliError already implements From<ConsumerError>

Reply:
```
pub fn invalid_arg<M: Into<String>>(reason: M) -> Self {
         Self::InvalidArg(reason.into())
     }
```

Reply:
Do you mean to implement `From<String> for ConsumerError` instead? I try to avoid implementing `From<String>` for errors because it can be easy to accidentally throw a `?` after a String value and turn an actual value into an error by mistake. I don't really see a big problem with having this function here, plus that's already how it was written, I didn't add this, I just moved it.

Reply:
fair enough